velox/dwio/common/ColumnVisitors.h

```diff
@@ -1516,24 +1531,47 @@ class DirectRleColumnVisitor
 template <bool kEncodingHasNulls>
 class StringColumnReadWithVisitorHelper {
  public:
-  StringColumnReadWithVisitorHelper(SelectiveColumnReader& reader, RowSet rows)
-      : reader_(reader), rows_(rows) {}
+  StringColumnReadWithVisitorHelper(
+      SelectiveColumnReader& reader,
+      RowSet rows,
+      bool hasDictionary)
```

Comment:
This should be a template parameter `bool kHasDictionary`

Reply:
Already refactored